calico_node/startup/startup_test.go

```diff
@@ -52,3 +59,173 @@ var _ = Describe("Non-etcd related tests", func() {
 		})
 	})
 })
+
+type EnvItem struct {
+	key string
+	value string
+}
+
+var _ = Describe("Etcd required tests", func() {
+	changedEnvVars := []string{"CALICO_IPV4POOL_CIDR", "CALICO_IPV6POOL_CIDR", "NO_DEFAULT_POOLS"}
+
+	BeforeEach(func() {
+		for _, envName := range changedEnvVars {
+			os.Unsetenv(envName)
+		}
+	})
+	AfterEach(func() {
+		for _, envName := range changedEnvVars {
+			os.Unsetenv(envName)
+		}
+	})
+
+	DescribeTable("Test IP pool env variables",
+		func(envList []EnvItem, expectPools bool, expectedIPv4 string, expectedIPv6 string) {
+
+			// Erase etcd clean.
+			testutils.CleanEtcd()
+
+			// Create a new client.
+			c, err := testutils.NewClient("")
+			if err != nil {
+				log.Println("Error creating client:", err)
+			}
+
+			// Set the env variables specified.
+			for _, env := range envList {
+				os.Setenv(env.key, env.value)
+			}
+			poolList, err := c.IPPools().List(api.IPPoolMetadata{})
+			Expect(poolList.Items).To(BeEmpty())
+
+			// Run the UUT.
+			configureIPPools(c)
+
+			// Get the IPPool list.
+			poolList, err = c.IPPools().List(api.IPPoolMetadata{})
+			Expect(err).NotTo(HaveOccurred())
+			log.Println("Get pool list returns: ", poolList.Items)
+
+			if expectPools {
+				// Look through the pool for the expected data.
+				foundv4Expected := false
+				foundv6Expected := false
+				for _, pool := range poolList.Items {
+					if pool.Metadata.CIDR.String() == expectedIPv4 {
+						foundv4Expected = true
+					}
+					if pool.Metadata.CIDR.String() == expectedIPv6 {
+						foundv6Expected = true
+					}
+				}
+				Expect(foundv4Expected).To(BeTrue(),
+					"Expected %s to be in Pools", expectedIPv4)
+				Expect(foundv6Expected).To(BeTrue(),
+					"Expected %s to be in Pools", expectedIPv6)
+			} else {
+				Expect(poolList.Items).To(BeEmpty(), "Environment %#v", os.Environ())
+			}
+		},
+
+		Entry("No env variables set", []EnvItem{}, true,
+			 "192.168.0.0/16", "fd80:24e2:f998:72d6::/64"),
+		Entry("IPv4 Pool env var set",
+			[]EnvItem{{"CALICO_IPV4POOL_CIDR", "172.16.0.0/24"}}, true,
+			"172.16.0.0/24", "fd80:24e2:f998:72d6::/64"),
+		Entry("IPv6 Pool env var set",
+			[]EnvItem{{"CALICO_IPV6POOL_CIDR", "fdff:ffff:ffff:ffff:ffff::/80"}}, true,
+			"192.168.0.0/16", "fdff:ffff:ffff:ffff:ffff::/80"),
+		Entry("NO_DEFAULT_POOLS env var set",
+			[]EnvItem{{"NO_DEFAULT_POOLS", "true"}}, false,
+			"", ""),
+		Entry("Both IPv4 and IPv6 Pool env var set",
+			[]EnvItem{{"CALICO_IPV4POOL_CIDR", "172.16.0.0/24"},{"CALICO_IPV6POOL_CIDR", "fdff:ffff:ffff:ffff:ffff::/80"}}, true,
+			"172.16.0.0/24", "fdff:ffff:ffff:ffff:ffff::/80"),
+	)
+
+	Describe("Test NO_DEFAULT_POOLS env variable", func() {
```

Comment:
It was not different.  I had split out that to be its own Describe section to remove the special case from the Table test because I thought that was cleaner, and then I forgot to go back and clean up the table test and remove the NO_DEFAULT_POOLS table entry.  I've cleaned up the table as I intended to now.